lean4-syntax.el

```diff
@@ -40,7 +40,7 @@
              (one-or-more digit) (optional (and "." (zero-or-more digit)))
              word-end)))
 
-(defconst lean4-warnings '("sorry" "exit" "admit") "lean warnings")
+(defconst lean4-warnings '("sorry") "lean warnings")
```

Comment:
I left `sorry` in here as it's definitely a keyword and highlighting it early doesn't hurt